lib/src/report/dependencies.dart

```diff
@@ -292,10 +292,18 @@ Future<List<OutdatedVersionDescription>> computeOutdatedVersions(
   }
   final result = <OutdatedVersionDescription>[];
 
-  final versionListing = jsonDecode(await getVersionListing(
-    name,
-    pubHostedUrl: hostedDependency.hosted?.url,
-  ));
+  Uri? environmentUri;
+  try {
+    // If 'PUB_HOSTED_URL' is not null, use it first
+    final hostUrlString = context.toolEnvironment.environment['PUB_HOSTED_URL'];
+    if (hostUrlString != null) {
+      environmentUri = Uri.parse(hostUrlString);
```

Comment:
`environmentUri = Uri.tryParse(hostUrlString);`
and let's drop the try-catch, as we in general don't like generic catch blocks, and the `environmentUri = null` is not needed either.